### PR TITLE
[iOS][macOS] Truncate URLs in log messages to prevent unbounded output

### PR DESCRIPTION
**Pull request by mallexxx**
Task/Issue URL: https://app.asana.com/1/137249556945/project/414235014887631/task/1216062097937483
Tech Design URL: N/A
CC: N/A

### Description
- Add `String.truncated(to:position:ellipsis:trimmingWhitespace:)` utility to `SystemFrameworksExtensions` with support for middle and tail truncation
- Add `URL.shortDescription` computed property to BSK returning `absoluteString` truncated to at most 1024 characters using middle ellipsis
- Replace `url.absoluteString` with `url.shortDescription` in log and debug messages across iOS, macOS, BSK (navigation, networking, history, bookmarks), WebExtensions, and DataBrokerProtectionCore to prevent unbounded URL strings in system logs
- Remove duplicate `truncated(length:trailing:)` and `truncated(length:middle:)` helpers from macOS `StringExtension`, now superseded by the shared implementation

### Testing Steps
1. Build and run the iOS app; open various pages and verify no regressions in navigation
2. Build and run the macOS app; open various pages and verify no regressions in navigation
3. Open a URL longer than 1024 characters and confirm log messages in console show the truncated form with `…` — e.g. `data:text/html,<html><body><h1>Test</h1><p>AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAaaaerjtceorpigjpserioguhcsmepgjcperighmsepgsjeprtimseprogitjmeportuihdmesportigAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA?abb=1&ssdd=34#fragment</p></body></html>`

### Impact and Risks
**Impact Level: None**

#### What could go wrong?
- Log messages now show truncated URLs for very long URLs — only affects debuggability of edge-case URLs, not user-visible behaviour
- No functional code paths changed; only logging output is affected

### Quality Considerations
- Unit tests cover all truncation modes (middle/tail), edge cases (max-length = 1, empty tail), whitespace trimming, and pass-through for short strings
- The 1024-character limit preserves enough URL context for debugging while eliminating runaway log entries from data-URIs or extremely long query strings
- No privacy implications: `shortDescription` does not change what data is logged, only its length

### Notes to Reviewer
All changes are mechanical substitutions of `absoluteString` → `shortDescription` at log call sites. The only non-trivial additions are the `String.truncated` implementation in `SystemFrameworksExtensions` and its tests.

Made with [Cursor](https://cursor.com)

### CODE REVIEW

File: macOS/DuckDuckGo/Common/Extensions/StringExtension.swift [hunk 1]
```diff
@@ -26,19 +26,6 @@ extension String {
 
     // MARK: - General
 
-    func truncated(length: Int, trailing: String = "…") -> String {
-      return (self.count > length) ? self.prefix(length) + trailing : self
-    }
```
Comment (diegoreymendez) [inline macOS/DuckDuckGo/Common/Extensions/StringExtension.swift:31]: We now have a similar method in iOS that's also dead code.  It's in `iOS/Core/StringExtension.swift:25`.

**Review by diegoreymendez — APPROVED** [review]
Approved.

Just a note that I didn't test "Trigger bookmark favicon fetching and confirm logs remain well-formed" because I've no idea what logs to look at or what well formed means... if it's important to re-test please let me know what I should look for.
